### DISCUSSION TITLE
Alternative json model depending on initial answer

### DISCUSSION THREAD

**Discussion by DiegoSVilla** [outlines-dev/outlines#244] [General]
Sry if this doesn't make sense or already has a possible implementation. But in the possibility that we have 3 possible json models as functions a model can run:

```
{name:regularResponse, content:.....}
{name:lights, lightArray:[], stateArray[(On, Off)]}
{name:heat, temperature:int}
```

for what I understand, this mask works linearly, so it would be possible to know the name and generate the following masks acordingly. Is this possible with pydantic right now? Or would it be only acheivable via regex?

**Comment by rlouf** [comment]
Are you saying that the JSON you want to generate would be conditional on the value that's generated for "name"? Or did I misunderstand?

**Reply by DiegoSVilla** [reply]
Yes! It would be a way for an assistant to choose a json format to respond, and still be guided through the process... maybe there is a better way?
